File: Yandex/Practicum/Algorightms_and_Data_Structures/8_sprint/B._Пограничный_контроль/solution.py
import sys


def solution() -> None:
    """
    Time Complexity O(N)
    Memory Complexity O(1)
    """
    a: str = sys.stdin.readline().rstrip()
    b: str = sys.stdin.readline().rstrip()
    n: int = len(a)
    m: int = len(b)
    if abs(n - m) > 1:
        sys.stdout.write('FAIL\n')
        return
    if n < m:
        a, b = b, a
        n, m = m, n
    left: int = 0
    right: int = 0
    already_skipped: bool = False
    while left < n and right < m:
        if a[left] == b[right]:
            left += 1
            right += 1
        elif not already_skipped:
            already_skipped = True
            if n != m: left += 1
            else:
                left += 1
                right += 1
        else:
            sys.stdout.write('FAIL\n')
            return
    sys.stdout.write(['FAIL', 'OK'][(left == n and right == m) or not already_skipped] + '\n')


if __name__ == '__main__':
    solution()

File: Yandex/Practicum/Algorightms_and_Data_Structures/8_sprint/B._Пограничный_контроль/solution.cpp
#include <iostream>
#include <string>
#include <cmath>


void solution() {
    /*
    Time Complexity O(N)
    Memory Complexity O(1)
    */
    std::string a, b;
    std::cin >> a >> b;
    int n = a.size(), m = b.size();
    if (std::abs(n - m) > 1) {
        std::printf("FAIL\n");
        return;
    }
    if (n < m) {
        std::swap(a, b);
        std::swap(n, m);
    }
    bool alreadySkipped = false;
    int left = 0, right = 0;
    while (left < n && right < m) {
        if (a[left] == b[right]) {
            ++left;
            ++right;
        } else if (!alreadySkipped) {
            if (n != m) ++left;
            else {
                ++left;
                ++right;
            }
            alreadySkipped = true;
        } else {
            std::printf("FAIL\n");
            return;
        }
    }
    std::cout << ((left == n && right == m) || !alreadySkipped? "OK" : "FAIL") << "\n";

}


int main() {
    solution();
}